Mathlib/Data/Nat/Factorial/Basic.lean

```diff
@@ -27,7 +27,6 @@ This file defines the factorial, along with the ascending and descending variant
 namespace Nat
 
 /-- `Nat.factorial n` is the factorial of `n`. -/
-@[simp]
```

Comment:
I think it's good that you remove this `@[simp]`, but please make `factorial_succ` a simp-lemma (18 lines below). Hopefully that means you don't have to make much changes to the rest of mathlib.

Reply:
Are you sure we want to make `factorial_succ` simp? I removed it here: #6840 and for this one there actually a place where we excluded it deliberately from firing.
I guess the question boils down to: is replacing `n.succ !`  by `(n + 1) * n !` really a simplicification most of the time. 
Since this means it having some lemma with `n.succ !` in it is not simpNF, I would argue in favor of not adding `@[simp]` back to `factorial_succ`.

Reply:
pinging also @alreadydone, since they originally suggested to remove simp from `factorial_succ`.

Reply:
@jcommelin what do you think? Does this need more discussion or can we go ahead with removing `@simp` here?

Reply:
Ok, fair enough. So let's not have `@[simp]` on the def nor on the lemma.